src/python/bcc/table.py

```diff
@@ -397,6 +397,39 @@ def clear(self):
         for k in self.keys():
             self.__delitem__(k)
 
+    def items_lookup_batch(self):
+        # batch size is set to the maximum
+        batch_size = self.max_entries
+        out_batch = ct.c_uint32(0)
+        keys = (type(self.Key()) * batch_size)()
+        values = (type(self.Leaf()) * batch_size)()
+        count = ct.c_uint32(batch_size)
+        res = lib.bpf_lookup_batch(self.map_fd,
+                                   None,
+                                   ct.byref(out_batch),
+                                   ct.byref(keys),
+                                   ct.byref(values),
+                                   ct.byref(count)
+                                   )
+
+        errcode = ct.get_errno()
+        if (errcode == errno.EINVAL):
+            raise Exception("BPF_MAP_LOOKUP_BATCH is invalid.")
+
+        if (res != 0 and errcode != errno.ENOENT):
+            raise Exception("BPF_MAP_LOOKUP_BATCH has failed")
+
+        for i in range(0, count.value):
+            yield (keys[i], values[i])
+
+    def items_delete_batch(self):
```

Comment:
I would like items_delete_patch takes an optional argument which is an array of keys. If users provide this array, we should be call self.items_delete_batch to delete these keys only. If user provides no keys, we can call items_lookup_and_delete_batch() to delete all of keys in the map. WDYT?

Reply:
Yes, good idea.

Reply:
And to be consistent, I should also provide the same optional argument, an array of keys, for items_lookup_and_delete_batch() and items_lookup_batch(). The src code should be almost the same than items_delete_batch(). If you are Ok with this commit https://github.com/iovisor/bcc/commit/4dbaeef03d284dc0ed931a918a36d69e646521cd, I will continue with the 2 other functions.

Reply:
> And to be consistent, I should also provide the same optional argument, an array of keys, for items_lookup_and_delete_batch() and items_lookup_batch(). The src code should be almost the same than items_delete_batch(). If you are Ok with this commit [4dbaeef](https://github.com/iovisor/bcc/commit/4dbaeef03d284dc0ed931a918a36d69e646521cd), I will continue with the 2 other functions.

@egobillot The new change looks good to me. Agree that let us make API consistent to have keys for batched lookup, lookup_and_delete as well. Thanks for working on this!